azurerm/internal/services/containers/kubernetes_nodepool.go

```diff
@@ -372,6 +383,11 @@ func FlattenDefaultNodePool(input *[]containerservice.ManagedClusterAgentPoolPro
 		}
 	}
 
+	criticalAddonsEnabled := false
+	if agentPool.NodeTaints != nil {
+		criticalAddonsEnabled = true
```

Comment:
this isn't strictly true that this is enabled by virtue of their being a taint (incase someone cycles the default node pool outside of Terraform, which we don't support at this point, but is a longer term thing to solve) - instead we can guarantee this by checking the node taints contain this value:

```suggestion
		for _, taint := range *agentPool.NodeTaints {
			if strings.EqualsFold(taint, "CriticalAddonsOnly=true:PreferNoSchedule") {
				criticalAddonsEnabled = true
			}
		}
```

Reply:
I thought that I wanted to get out of this too easy :) Good call, thank you.